### PR TITLE
Set repo CI check requirements

### PR DESCRIPTION
**Pull request by wu-sheng**
@kezhenxu94 Do I use the correct CI task name?

### CODE REVIEW

**Comment by wu-sheng** [comment]
@hanahmily I still keep the `main` branch unprotected, direct push is still working.

File: .asf.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    rebase:  false
+  protected_branches:
+    main:
+      required_status_checks:
+        strict: true
+        contexts:
+          - Continuous Integration
+      required_pull_request_reviews:
+        dismiss_stale_reviews: true
+        required_approving_review_count: 1
```
Comment (kezhenxu94) [inline .asf.yaml:39]: > @hanahmily I still keep the main branch unprotected, direct push is still working.

This is not the case you said

**Review by kezhenxu94 — APPROVED** [review]

File: .asf.yaml (hunk 1 above)
Comment (wu-sheng) [reply .asf.yaml:39]: Is this going to add branch protection? This was for review requirement

Comment (kezhenxu94) [reply .asf.yaml:39]: > Is this going to add branch protection? This was for review requirement

This config `strict: true` protects the branch.

**Comment by hanahmily** [comment]
What I concern is do I have the legacy to merge any PRs?

File: .asf.yaml (hunk 1 above)
Comment (kezhenxu94) [reply .asf.yaml:39]: > > Is this going to add branch protection? This was for review requirement
> 
> This config `strict: true` protects the branch.

And this

```
contexts:
  - Continuous Integration
```

According to GitHub

> - [x] Require status checks to pass before merging
> Choose which status checks must pass before branches can be merged into a branch that matches this rule. When enabled, commits must first be pushed to another branch, then merged or pushed directly to a branch that matches this rule after status checks have passed.

**Comment by wu-sheng** [comment]
You could merge others' PRs, not yourself.

**Comment by wu-sheng** [comment]
@hanahmily @kezhenxu94 I removed the review requirement for now. It should be added back once we are closing to our first release. We need to be careful after that.

File: .asf.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    - database
+  enabled_merge_buttons:
+    squash:  true
+    merge:   false
+    rebase:  false
+  protected_branches:
+    main:
+      required_status_checks:
+        contexts:
+          - Continuous Integration
```
Comment (kezhenxu94) [inline .asf.yaml:35]: These still not allows direct push to main branch

Comment (wu-sheng) [reply .asf.yaml:35]: Yes, but they can merge the PR through pull request alternatively.

Comment (kezhenxu94) [reply .asf.yaml:35]: If we don't want to allow direct push to main branch but allow merging committers' own PRs, this is OK to me

Comment (wu-sheng) [reply .asf.yaml:35]: Currently require CI passed and a committer merge(squash only).

**Review by kezhenxu94 — APPROVED** [review]

**Pull request merged**